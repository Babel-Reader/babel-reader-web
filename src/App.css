.react-pdf__Page__canvas {
    margin:auto;
    position: absolute;
    right: 0;
    left: 0;
}


.react-pdf__Page__textContent {
    height: 100% !important;
    position: absolute !important;
    transform: none !important;
    left: 0 !important;
    right: 0 !important;
    margin: auto !important;
}

