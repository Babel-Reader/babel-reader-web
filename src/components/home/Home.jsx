import React, {useState} from 'react';
import {withRouter} from "react-router-dom";
import Reading from "../reading/Reading";
import './Home.scss'
import Header from "../header/Header";
import {languageList} from "../../utils";
import Footer from "../footer/Footer";

export const BookContext = React.createContext()

export default withRouter(()=>{
  const [file, setFile] = useState("ALICE'S ADVENTURES IN WONDERLAND.pdf");
  const [pageNb, setPageNb] = useState(1);
  const [pageCount, setPageCount] = useState();
  const [languages, setLanguages] = useState({
    in: {
      key: 'auto'
    },
    out: languageList[0]
  })

  return (
    <BookContext.Provider value={{
      file,
      setFile,
      pageNb,
      setPageNb,
      pageCount,
      setPageCount,
      languages,
      setLanguages
    }}>
      <section className="container">
        <Header />
        <Reading/>
        <Footer {...{}}/>

      </section>
    </BookContext.Provider>
  );
})
