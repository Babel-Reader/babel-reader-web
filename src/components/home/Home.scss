.container {
  display: flex;
  flex-direction: column;
  position: absolute;
  height: 100%;
  width: 100%;
  overflow: hidden;

  .reading-container {
    max-width: 100vw;
  }
}
