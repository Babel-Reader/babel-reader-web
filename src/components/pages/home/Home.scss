.container {
  display: flex;
  flex-direction: column;
  position: absolute;
  height: 100%;
  width: 100%;
  overflow: hidden;
  justify-content: space-between;

  .reading-container {
    max-width: 100vw;
  }
}
