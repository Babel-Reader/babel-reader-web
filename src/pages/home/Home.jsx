import React from 'react';
import {Link, withRouter} from "react-router-dom";
import {useDropzone} from 'react-dropzone'
import Button from "@material-ui/core/Button";

export default withRouter(()=>{

  const {acceptedFiles, getRootProps, getInputProps} = useDropzone({accept:'.pdf'});

  const files = acceptedFiles.map(file => (
    <li key={file.path}>
      {file.path} - {file.size} bytes
    </li>
  ));

  return (
    <section className="container">
      <div {...getRootProps({className: 'dropzone'})}>
        <input {...getInputProps()} />
        <p>Drag 'n' drop some files here, or click to select files</p>
      </div>
      <aside>
        <h4>Files</h4>
        <ul>{files}</ul>
      </aside>
      <Button component={Link} to={`/read`} disabled={!files.length}>
        sample.pdf
      </Button>

    </section>
  );
})
