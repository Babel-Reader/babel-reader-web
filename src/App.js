import React, { useEffect, useState } from 'react';
import './App.css';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Home from './components/pages/home/Home';
import { pdfjs } from 'react-pdf';
import Library from './components/pages/library';
import { bookTitle } from 'utils';
import { auth } from 'services/firebase/firebase';

pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;

export const LANG_AUTO = {
  key: '??',
  name: 'auto detect',
  nativeName: 'auto detect',
};

let languageList = [
  LANG_AUTO,
  {
    key: 'en',
    name: 'English',
    nativeName: 'English',
  },
  {
    key: 'fr',
    name: 'French',
    nativeName: 'Français',
  },
  {
    key: 'es',
    name: 'Spanish',
    nativeName: 'Español',
  },
];

const fetchLanguageList = () => {
  return fetch(process.env.REACT_APP_TRANSLATE_LANGUAGE_LIST_URL)
    .then((res) => res.json())
    .then((res) => {
      if (res && res.translation) {
        const langValueForKey = (key) => {
          return {
            key,
            name: res.translation[key].name,
            nativeName: res.translation[key].nativeName,
          };
        };

        languageList = [
          LANG_AUTO,
          langValueForKey('en'),
          langValueForKey('fr'),
          langValueForKey('es'),
        ];
        Object.keys(res.translation).forEach((key) => {
          if (key !== 'en' && key !== 'fr' && key !== 'es') {
            languageList.push(langValueForKey(key));
          }
        });
      }
    });
};

export const BookContext = React.createContext({});

function App() {
  const [file, setFile] = useState();
  const [user, setUser] = useState(auth.currentUser);
  const [languages, setLanguages] = useState({
    in: LANG_AUTO,
    out: languageList[2],
  });

  useEffect(() => {
    const title = bookTitle(file);
    const pageTitle = 'Babel Reader';
    document.title = title ? `${pageTitle}: ${title}` : pageTitle;
  }, [file]);

  useEffect(() => {
    fetchLanguageList().then(() => {
      setLanguages({
        in: LANG_AUTO,
        out: languageList[2],
      });
    });
  }, []);

  return (
    <BookContext.Provider
      value={{
        file,
        setFile,
        languages,
        setLanguages,
        languageList,
        user,
        setUser,
      }}
    >
      <Router>
        <Switch>
          <Route path="/reading/samples/:bookName">
            <Home isSample />
          </Route>
          <Route path="/reading/:bookName">
            <Home />
          </Route>
          <Route>
            <Library />
          </Route>
        </Switch>
      </Router>
    </BookContext.Provider>
  );
}

export default App;
