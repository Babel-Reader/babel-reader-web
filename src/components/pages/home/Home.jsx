import React from 'react';
import Reading from './reading/Reading';
import './Home.scss';
import Header from './header/Header';
import Footer from './footer/Footer';


export default () => {

  return (
    <section className="container">
      <Header/>
      <Reading/>
      <Footer {...{}} />
    </section>
  );
};
