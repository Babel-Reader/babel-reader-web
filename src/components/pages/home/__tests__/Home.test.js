import React from 'react';
import { create } from 'react-test-renderer';
import { BookContext } from 'App';
import Home from 'components/pages/home';
import { MemoryRouter } from 'react-router-dom';

let languageList = [
  {
    key: 'auto',
    name: 'auto detect',
    nativeName: 'auto detect',
  },
  {
    key: 'en',
    name: 'English',
    nativeName: 'English',
  },

  {
    key: 'fr',
    name: 'French',
    nativeName: 'Français',
  },
  {
    key: 'es',
    name: 'Spanish',
    nativeName: 'Español',
  },
];
it('can match snapshot', () => {
  expect(
    create(
      <BookContext.Provider
        value={{
          file: 'testFile',
          languageList,
          languages: {
            in: languageList[0],
            out: languageList[2],
          },
        }}
      >
        <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
          <Home/>
        </MemoryRouter>
      </BookContext.Provider>,
    ),
  ).toMatchSnapshot();
});
