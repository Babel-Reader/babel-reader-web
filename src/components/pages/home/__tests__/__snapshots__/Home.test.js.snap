// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can match snapshot 1`] = `
<section
  className="container"
>
  <header
    className="home-header"
  >
    <section
      className="dropzone"
      onBlur={[Function]}
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <input
        accept=".pdf"
        autoComplete="off"
        multiple={true}
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
        type="file"
      />
      <div />
      <div
        className="header-title"
      >
        <p
          className="header-book-title"
        >
          ALICE'S ADVENTURES IN WONDERLAND.pdf
        </p>
        <i>
          Click to open a book
        </i>
      </div>
    </section>
    <div
      className="lang-input"
    >
      <div
        aria-expanded={false}
        aria-owns={null}
        className="MuiAutocomplete-root lang-combo-box MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            id="undefined-label"
          >
            Inlang
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={false}
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Clear"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-expanded={false}
        aria-owns={null}
        className="MuiAutocomplete-root lang-combo-box MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            id="undefined-label"
          >
            Outlang
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            onClick={[Function]}
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls={null}
              aria-invalid={false}
              autoCapitalize="none"
              autoComplete="off"
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              required={false}
              spellCheck="false"
              type="text"
              value=""
            />
            <div
              className="MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Clear"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Clear"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Open"
                className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Open"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="reading-container"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text nav-btn-back Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        &lt;
      </span>
    </button>
    <div
      className="react-pdf__Document reading-document"
    >
      <div
        className="react-pdf__message react-pdf__message--loading"
      >
        Loading PDF…
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text nav-btn-forward"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        &gt;
      </span>
    </button>
  </div>
  <footer
    className="home-footer"
  >
    <span
      className="MuiSlider-root MuiSlider-colorPrimary"
      onMouseDown={[Function]}
    >
      <span
        className="MuiSlider-rail"
      />
      <span
        className="MuiSlider-track"
        style={
          Object {
            "left": "0%",
            "width": "49.74874371859296%",
          }
        }
      />
      <input
        type="hidden"
        value="100"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax={200}
        aria-valuemin={1}
        aria-valuenow={100}
        className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-1"
        data-index={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        role="slider"
        style={
          Object {
            "left": "49.74874371859296%",
          }
        }
        tabIndex={0}
      >
        <span
          className="PrivateValueLabel-offset-3 MuiSlider-valueLabel"
        >
          <span
            className="PrivateValueLabel-circle-4"
          >
            <span
              className="PrivateValueLabel-label-5"
            >
              100
            </span>
          </span>
        </span>
      </span>
    </span>
    <div
      className="page-input"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              style={
                Object {
                  "textAlign": "right",
                }
              }
              type="text"
              value={1}
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-7"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </form>
      <div
        className="page-input-total"
      >
        
      </div>
    </div>
  </footer>
</section>
`;
